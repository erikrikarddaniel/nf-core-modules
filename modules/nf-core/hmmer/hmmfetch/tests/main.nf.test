nextflow_process {

    name "Test Process HMMER_HMMFETCH"
    script "../main.nf"
    process "HMMER_HMMFETCH"

    tag "modules"
    tag "modules_nfcore"
    tag "hmmer"
    tag "hmmer/hmmfetch"

    test("hmmfetch - key") {

        when {
            process {
                """
                
                input[0] = [
                    [ id:'test', single_end:false ],
                    file('https://raw.githubusercontent.com/tseemann/barrnap/master/db/arc.hmm')
                ]
                input[1] = Channel.of('16S_rRNA').first()
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match('key_out') },
                { assert snapshot(process.out.hmm).match('key_hmm') },
                { assert snapshot(process.out.versions).match('key_versions') }
            )
        }

    }

    test("hmmfetch - no overwrite") {

        when {
            process {
                """
                
                input[0] = [
                    [ id:'arc', single_end:false ],
                    file('https://raw.githubusercontent.com/tseemann/barrnap/master/db/arc.hmm')
                ]
                input[1] = Channel.of('16S_rRNA').first()
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match('nooverwrite_out') },
                { assert snapshot(process.out.hmm).match('nooverwrite_hmm') },
                { assert snapshot(process.out.versions).match('nooverwrite_versions') }
            )
        }

    }

    test("hmmfetch - keyfile") {

        when {
            process {
                """
                
                input[0] = [
                    [ id:'test', single_end:false ],
                    file('https://raw.githubusercontent.com/tseemann/barrnap/master/db/arc.hmm')
                ]
                input[1] = []
                input[2] = Channel
                    .of('16S_rRNA', '23S_rRNA')
                    .collectFile(name: 'keys.txt', newLine: true)
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match('keyfile_out') },
                { assert snapshot(process.out.hmm).match('keyfile_hmm') },
                { assert snapshot(process.out.versions).match('keyfile_versions') }
            )
        }

    }

    test("hmmfetch - index") {

        when {
            process {
                """
                
                input[0] = [
                    [ id:'test', single_end:false ],
                    file('https://raw.githubusercontent.com/tseemann/barrnap/master/db/arc.hmm')
                ]
                input[1] = []
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match('index_out') },
                { assert snapshot(process.out.index).match('index_index') },
                { assert snapshot(process.out.versions).match('index_versions') }
            )
        }

    }

}
