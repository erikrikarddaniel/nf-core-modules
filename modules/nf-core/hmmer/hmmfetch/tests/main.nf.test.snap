{
    "key_hmm": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.hmm:md5,0328406a7469af6f24e2157b0367705f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T09:50:10.639712907"
    },
    "keyfile_hmm": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.hmm:md5,0e5c7e77342ec91bc6fac2a7a0228d08"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T09:59:41.836554457"
    },
    "index_out": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "arc.hmm.ssi:md5,17a7bb35662baabb5b1474660d072fb6"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ],
                "hmm": [
                    
                ],
                "index": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "arc.hmm.ssi:md5,17a7bb35662baabb5b1474660d072fb6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T10:07:00.226687276"
    },
    "key_out": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmm:md5,0328406a7469af6f24e2157b0367705f"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ],
                "hmm": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmm:md5,0328406a7469af6f24e2157b0367705f"
                    ]
                ],
                "index": [
                    
                ],
                "versions": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T09:50:10.623341213"
    },
    "index_index": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "arc.hmm.ssi:md5,17a7bb35662baabb5b1474660d072fb6"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T10:07:00.237979636"
    },
    "nooverwrite_hmm": {
        "content": [
            [
                [
                    {
                        "id": "arc",
                        "single_end": false
                    },
                    "arc.out.hmm:md5,0328406a7469af6f24e2157b0367705f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T10:23:56.556811704"
    },
    "nooverwrite_versions": {
        "content": [
            [
                "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T10:23:56.575136343"
    },
    "keyfile_versions": {
        "content": [
            [
                "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T09:59:41.874358057"
    },
    "keyfile_out": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmm:md5,0e5c7e77342ec91bc6fac2a7a0228d08"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ],
                "hmm": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmm:md5,0e5c7e77342ec91bc6fac2a7a0228d08"
                    ]
                ],
                "index": [
                    
                ],
                "versions": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T09:59:41.786988564"
    },
    "index_versions": {
        "content": [
            [
                "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T10:07:00.247943543"
    },
    "nooverwrite_out": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "arc",
                            "single_end": false
                        },
                        "arc.out.hmm:md5,0328406a7469af6f24e2157b0367705f"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ],
                "hmm": [
                    [
                        {
                            "id": "arc",
                            "single_end": false
                        },
                        "arc.out.hmm:md5,0328406a7469af6f24e2157b0367705f"
                    ]
                ],
                "index": [
                    
                ],
                "versions": [
                    "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T10:23:56.536863962"
    },
    "key_versions": {
        "content": [
            [
                "versions.yml:md5,ca18131840322abff960ba5c09f91a5a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-08T09:50:10.64971829"
    }
}